cloudformation/template.yml

```diff
@@ -208,11 +220,41 @@ Parameters:
       A list of IDs of subnets in which to launch Batch instances (all subnets
       must exist in the VPC you selected)
 
+  GpuLaunchTemplate:
+    Type: String
+    Default: ""
+    Description:
+      (Optional) The ID of an EC2 Launch Template to associate with the GPU-based
+      Batch Compute Environment.
+
+  GpuLaunchTemplateVersion:
+    Type: Number 
+    Default: 1
+    Description:
+      (Optional) The version of the EC2 Launch Template to associate with the GPU-based
+      Batch Compute Environment. 
```

Comment:
I'm not sure how useful it will be to expose versioning for launch templates. On the one hand, exposing the version should hypothetically allow the user to update a stack to read from a new version if they update their template, which seems useful. On the other hand, while testing this I ran into some unexpected behavior where [CloudFormation won't update a custom-named resource](https://aws.amazon.com/premiumsupport/knowledge-center/cloudformation-custom-name/), which happens to include the compute environment in this module. That means that in order to update your stack without recreating it you'd have to do something hacky like rename the compute environment out of band before updating the stack (at which point you'll probably just want to skip it all and update the launch template version for the compute environment out of band).

In the ideal world we could just set this to pull the "default" version of the launch template, but [this blog post points out](https://www.kindlyops.com/knowledge-base/cloudformation-launch-templates/) that if no version number is specified CloudFormation will default to version `1`.

Reply:
Interesting. For the ECS cluster module in Terraform, there is a magic [`$$Latest`](https://github.com/azavea/terraform-aws-ecs-cluster/blob/9d7ff9af1c2807d55bb369296fc187a3a67d108a/main.tf#L192) that pulls the latest version of the launch template. It is unclear whether this is a magic variable evaluated by AWS, or Terraform itself.

It also looks like `AWS::EC2::LaunchTemplate` has a `LatestVersionNumber` attribute.

Would it be possible to default to the latest if `GpuLaunchTemplateVersion` is not specified?

Reply:
Interesting! I couldn't find any official documentation about how to retrieve the latest version, and since it defaults to version 1 I assumed that there wasn't one. I think it's worth giving `$$Latest` a try, however.